nncf/openvino/quantization/cache_statistics.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) 2024 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict, List, Tuple
+
+import openvino.runtime as ov
+
+from nncf.common.factory import NNCFGraphFactory
+from nncf.common.factory import StatisticsAggregatorFactory
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.graph.graph import NNCFNode
+from nncf.common.tensor_statistics.aggregator import StatisticsAggregator
+from nncf.data import Dataset
+from nncf.openvino.graph.model_utils import remove_friendly_name_duplicates
+from nncf.parameters import SensitivityMetric
+from nncf.quantization.algorithms.weight_compression.algorithm import WeightCompression
+from nncf.quantization.algorithms.weight_compression.algorithm import get_weight_compression_configuration
+from nncf.quantization.algorithms.weight_compression.gptq import GPTQ
+from nncf.quantization.algorithms.weight_compression.mixed_precision import MIXED_PRECISION_CRITERIA
+
+
+def register_statistics_for_algorithm(
+    statistics_aggregator: StatisticsAggregator,
+    model: ov.Model,
+    graph: NNCFGraph,
+    subset_size: int,
+    compression_algorithm: WeightCompression,
+) -> None:
+    """Registers the statistics of the provided algorithm."""
+    compression_algorithm.set_backend_entity(model)
+
+    nodes_to_compress = [
+        node
+        for node in compression_algorithm.get_nodes_to_compress(graph)
+        if node.metatype in compression_algorithm._backend_entity.matmul_metatypes
+    ]
+
+    matmul_input_to_output_nodes_map = compression_algorithm.get_matmul_input_to_output_nodes_map(
+        nodes_to_compress, graph
+    )
+
+    statistic_points = compression_algorithm.get_statistic_points(
+        model, graph, matmul_input_to_output_nodes_map.keys(), subset_size
+    )
+    statistics_aggregator.register_statistic_points(statistic_points)
+
+
+def _register_gptq(
```

Comment:
gptq statistics are not used after PR https://github.com/openvinotoolkit/nncf/pull/3003. I would suggest to exclude gptq statistics from the caching.

Reply:
disabled